aries_cloudagent/conductor.py

```diff
@@ -277,9 +284,40 @@ def __init__(self, context_builder: ContextBuilder) -> None:
             parsed_msg, delivery, connection, self.outbound_message_router
         )
         if socket:
+            # if a reply mode is present then a response is allowed over this connection
+            # wait till dispatch is complete,
+            # then check to see if any queued responses can be sent.
+            if socket.reply_mode:
+                await complete
+                await self.queue_processing(socket)
+
+            # close a single_response socket if no answer given by now.
```

Comment:
This method returns the async task so that it can be cancelled, which is why it uses a done callback instead of awaiting the task. The queue processing call could be added to the done callback.

Reply:
I didn't use an additional done callback because the docs were inspecific about execution order and the queue processing must be complete before the dispatch_complete() is called. 

Are you proposing that `queue_processing()` be called from inside `dispatch_complete()`? That feels like an odd flow to me.

Reply:
The method could return one task which awaits the dispatch then awaits `queue_processing`

Reply:
I'm not sure I follow what you are proposing here. Can you elaborate a little more?